### PR TITLE
bundle assets in slim loader

### LINKED ISSUE #825: bundleAsset test removed?
is there a reason why the bundleAsset test was removed?
in 1.x the test was removed, in 0.16 the test exists
https://github.com/stealjs/steal-tools/tree/2371ad6db30e78eac34f61275de112f28f1894a8/test/bundle_assets

### LINKED ISSUE #846: bundleAssets is not working with optimize build
@pYr0x commented on [Thu Sep 21 2017](https://github.com/stealjs/steal/issues/1274)

__How often can you reproduce it?__ 

- [x] Always
- [ ] Sometimes
- [ ] Rarely
- [ ] Unable
- [ ] I didn’t try

__Description:__
bundleAssets is not working with optimize build
```
let buildResult = await stealTools.optimize({
        config: path.join(__dirname, "package.json!npm")
    }, {
        bundleAssets: {
            glob: [
                path.join(__dirname, "src", "assets", "**", "*"),
                path.join(__dirname, "src", "styles.css"),
                path.join(__dirname, "src", "manifest.json")
            ]
        },
        minify: true,
        bundlePromisePolyfill: false,
        splitLoader: true
    });
```

__Expected results:__
assets are copied

__Actual results:__
assets are not copied into the dist folder

__Environment:__

| Software                     | Version
| --------------------- | -------
| Steal version              | latest
| Steal-tools version    | latest
| node -v                       | 8
| npm -v                        | 5
| Browser                      | chrome
| Operating system      | mac

### PR DESCRIPTION
**Pull request by pYr0x**
blocked by: https://github.com/stealjs/steal-bundler/pull/30

this PR integrates steal-serviceworker 
and close #846 and close #825 

the code i added can definitely refactored. 
```
// run external steal-tool plugins after the build
				if(options){

					var p = Promise.resolve(builtResult);

					if(options.bundleAssets){
						var bundleAssets = require("steal-bundler");
						p = p.then(function(builtResult) {
							return bundleAssets(builtResult, options.bundleAssets);
						});
					}

					if(options.serviceWorker){
						var precache = require("steal-serviceworker");
						p = p.then(function (builtResult) {
							return precache(builtResult, options.serviceWorker);
						});
					}
				}

				p.then(function (builtResult) {
					resolve(builtResult);
				}).catch(function (error) {
					reject(error);
				});
```

but for now we can live with that.

maybe we can create some API for post-plugins like steal-bundler, steal-serviceworker and steal-electon that need the buildResult.

cc @matthewp @m-mujica any thought how we can do this?

### CODE REVIEW

File: lib/build/multi.js [hunk 1]
```diff
@@ -109,21 +109,34 @@ module.exports = function(config, options){
...
-
-				// If bundleAssets is truthy run the bundler after the build.
-				if(options && options.bundleAssets) {
-					require("steal-bundler")(data, options.bundleAssets)
-						.then(function(){
-							resolve(data);
-						}, function(err){
-							reject(err);
+			writeStream.on("data", function(builtResult){
+				//this.end();
```
Comment (pYr0x) [inline lib/build/multi.js]: @matthewp dont know whats that function is doing. if i comment it out, it worked for the multibuild, but not for the slim build https://github.com/stealjs/steal-tools/pull/848/files#diff-b016b898f98b8eb1dc4d58c367f1bc38R99
is `this.end()` necessary?

**Comment by pYr0x** [comment]
the test does not pass because travis-ci runs steal-tools on node 4 and steal-serviceworker supports node 6+
@matthewp any possibility to drop support for node 4?
otherwise i have to transpile the serviceworker script to node 4

**Comment by matthewp** [comment]
We can't drop Node 4 until steal-tools 2.0. Looking at the build error: https://travis-ci.org/stealjs/steal-tools/jobs/279253395

I'm not sure it's a syntax problem... have you tried running it in Node 4 locally?

**Comment by pYr0x** [comment]
remove the integration of serviceworker because transpiling the source script to node 4 cause problems and i dont want to refactor steal-serviceworker.

this PR is still blocked by: stealjs/steal-bundler#30

File: lib/build/slim.js [hunk 2]
```diff
@@ -91,7 +92,37 @@ module.exports = function(cfg, opts) {
...
+
+				// steal-serviceworker can't be integrated until we drop support for node 4
+				// tests are written and ready
+				// maybe we come up with a better idea integrate post-build-plugins
+				// if(options.serviceWorker){
+				// 	var precache = require("steal-serviceworker");
+				// 	p = p.then(function (builtResult) {
+				// 		return precache(builtResult, options.serviceWorker);
+				// 	});
+				// }
```
Comment (matthewp) [inline lib/build/slim.js]: Same as above.

**Pull request merged**